aesop/spectra.py

```diff
@@ -79,6 +79,47 @@ def __init__(self, wavelength=None, flux=None, name=None, mask=None,
         self.time = time
         self.continuum_normalized = continuum_normalized
 
+    def flux_calibrate(self, fluxed_spec, polynomial_order, plots=False):
+        """
+        Interpolate high-res spectrum to low-res flux calibrated spectrum, then fit
+        the ratio with a polynomial to flux calibrate. Returns polynomial coefficients
+
+        Parameters
+        ----------
+        fluxed_spec : `~aesop.Spectrum1D`
+            Already flux calibrated low-resolution spectrum of the same object
+        polynomial_order : int
+            Order of polynomial fit
+        plots : bool
+            If True, plot the sensitivity data and the fit
+
+        Returns
+        -------
+        fit_params : `~numpy.ndarray`
+            Best-fit polynomial coefficients
+        """
+
+        int_spectrum = interpolate_spectrum(spectrum=self, new_wavelengths=fluxed_spec.wavelength)
+
+        sens_data = fluxed_spec.flux/int_spectrum.flux
+
+        fit_params = np.polyfit(int_spectrum.wavelength,sens_data, polynomial_order)
+
+        if plots:
+            plt.figure()
+            plt.plot(int_spectrum.wavelength,
+                     sens_data)
+            plt.plot(int_spectrum.wavelength,
+                     np.polyval(fit_params,
+                                int_spectrum.wavelength))
+            plt.show()
+
+        return fit_params
```

Comment:
If I were to see a function that looked like this: 

```python
flux_calibrated_spectrum = fits.getdata(path1)
target_spectrum = EchelleSpectrum.from_fits(path2)
target_spectrum.flux_calibrate(flux_calibrated_spectrum)
```
I think I would expect the `flux_calibrate` method to alter the `target_spectrum` object, rather than return the polynomial coefficients. Maybe this function should be called `flux_calibration_parameters`, and the `flux_calibrate` method should actually apply the polynomial to each order?

Reply:
Theoretically this step would be after merging all of the orders into a continuum-normalized spectrum; I can make a flux_calibrate method for ``EchelleSpectrum`` that does the merging, normalizing, then flux calibrating?

Reply:
Ah, sorry, my example should have used `Spectrum1D` instead of `EchelleSpectrum`.

Reply:
Can do. Would it make sense for ``flux_calibrate_parameters`` to also save the parameters of the fit to a ``sens_params`` attribute? Then if you want, you can separately run ``flux_calibrate_parameters`` and check how you did with ``plots=True``, before running ``flux_calibrate``, which will try to use ``sens_params`` if the attribute exists. Or you can just call ``flux_calibrate`` and give it the vars that ``flux_calibrate_parameters`` needs.

Reply:
Let's keep the parameters as an input/output rather than an attribute, so it'll always be crystal clear to the user where the parameters are coming from.

Reply:
Done and done.